jl4-core/src/L4/Utils/IntervalMap.hs

```diff
@@ -0,0 +1,53 @@
+module L4.Utils.IntervalMap (
+  IV.IntervalMap,
+  IV.search,
+  IV.dominators,
+  IV.insert,
+  IV.union,
+  IV.high,
+  IV.low,
+  IV.singleton,
+  IV.empty,
+  smallestContaining,
+  intervalToRange,
+  intervalToRange0,
+  srcRangeToInterval,
+  intervalToSrcRange,
+) where
+
+import Base
+
+import L4.Parser.SrcSpan
+
+import Control.Arrow (first)
+import HaskellWorks.Data.IntervalMap.FingerTree (Interval, IntervalMap)
+import qualified HaskellWorks.Data.IntervalMap.FingerTree as IV
+
+smallestContaining ::
+  NormalizedUri ->
+  SrcPos ->
+  IntervalMap SrcPos b ->
+  Maybe (SrcRange, b)
+smallestContaining nuri sp info =
+  let
+    r = IV.search sp info
+  in
+    case r of
+      [] -> Nothing
+      _ -> Just (first (intervalToRange0 nuri) $ maximumBy cmp r)
+ where
+  cmp (a, _) (b, _) =
+    (b.high `compare` a.high)
+      <> (a.low `compare` b.low)
```

Comment:
shouldn't this be just `listToMaybe $ sortOn (\i -> i.hi - i.lo) $ search sp info`? What additionaly thing does this do

Reply:
`SrcPos` does not implement `Num`, otherwise it is the same

Reply:
Reimplemented just like you proposed.